impl/src/main/java/slack/SlackResourceImpl.java

```diff
@@ -30,52 +37,111 @@ public SlackResourceImpl(SlackConfig slackConfig) {
         this.slackConfig = slackConfig;
     }
 
+    static String handleUserInfoResponse(UsersInfoResponse usersInfoResponse) {
+        if (usersInfoResponse.isOk()) {
+            if (usersInfoResponse.getUser().getProfile().getEmail() == null) {
+                LOG.warn("Could not fetch users email, maybe scope users:read.email is missing");
+                throw new IllegalStateException("User email from slack is null and might be due to missing scope users:read.email");
+            }
+            return usersInfoResponse.getUser().getProfile().getEmail();
+        } else {
+            throw new IllegalStateException("Could not fetch user email from slack: " + usersInfoResponse.getError());
+        }
+    }
+
+    static String handleUserByEmailResponse(UsersLookupByEmailResponse usersLookupByEmailResponse) {
+        if (!usersLookupByEmailResponse.isOk()) {
+            throw new IllegalStateException("Could not get user by email from slack: " + usersLookupByEmailResponse.getError());
+        }
+
+        return usersLookupByEmailResponse.getUser().getId();
+    }
+
+    static Observable<Void> handleChatPostMessageResponse(ChatPostMessageResponse chatPostMessageResponse) {
+        if (!chatPostMessageResponse.isOk()) {
+            LOG.warn("Could not post to slack: {}", chatPostMessageResponse.getError());
+            return error(new IllegalStateException(chatPostMessageResponse.getError()));
+        }
+        return empty();
+    }
+
+    static Message handleGetMessageResponse(ChannelsRepliesResponse channelsRepliesResponse) {
+        if (channelsRepliesResponse.isOk()) {
+            return channelsRepliesResponse.getMessages().get(0);
+        }
+        LOG.warn("Could not get message from slack: {}", channelsRepliesResponse.getError());
+        throw new IllegalStateException(channelsRepliesResponse.getError() + " " + channelsRepliesResponse.getWarning());
+    }
+
     @Override
     public Observable<String> getUserEmail(String userId) {
-        return fromCallable(() -> slack.methods().usersInfo(
-            UsersInfoRequest
-            .builder()
+        return callSlack(() -> {
+
+            final UsersInfoRequest slackRequest = UsersInfoRequest
+                .builder()
                 .user(userId)
                 .token(slackConfig.getApiToken())
-            .build())
-            .getUser()
-            .getProfile()
-            .getEmail());
+                .build();
+
+            return slack.methods().usersInfo(slackRequest);
```

Comment:
the runnable will hide the checked exceptions thrown from the usersInfo method. not a issue but good to know maybe.

Reply:
errors that will result in no response, will bubble up. It might be hard to figure out by reading the code, because we log errors here that results in a response. But im not sure what to do about that. Maybe leave that for now. Maybe its fine.